File: assignments/a1/q9/Java/Optional.java
//Optional:top//
//Optional:import//
import java.util.*;

// <token_rule>:Optional ::= TOKEN <RULE>
public class Optional extends Token_rule /*Optional:class*/ {

    public static final String $className = "Optional";
    public static final String $ruleString =
        "<token_rule>:Optional ::= TOKEN <RULE>";

    public Token rule;

    public Optional(Token rule) {
//Optional:init//
        this.rule = rule;
    }

    public static Optional parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<token_rule>:Optional", scn$.lno);
        scn$.match(Token.Match.TOKEN, trace$);
        Token rule = scn$.match(Token.Match.RULE, trace$);
        return new Optional(rule);
    }

//Optional//
}

File: assignments/a1/q9/Python/Optional.py
#Optional:top#
#Optional:import#

# <token_rule>:Optional ::= TOKEN <RULE>
class Optional(Token_rule): #Optional:class#

    className = "Optional"
    ruleString = "<token_rule>:Optional ::= TOKEN <RULE>"
    rule = None

    def __init__(rule):
        #Optional:init#
        self.rule = rule

#Optional#
